src/SignalR/clients/ts/signalr/src/HubConnection.ts

```diff
@@ -127,6 +127,27 @@ export class HubConnection {
         return this.connection ? (this.connection.connectionId || null) : null;
     }
 
+    /** Indicates the url of the {@link HubConnection} to the server. */
+    get baseUrl(): string {
+        return this.connection.baseUrl || "";
+    }
+
+    /**
+     * Sets a new url for the HubConnection.
```

Comment:
Should have some text explaining that this should only be set when the connection isn't active.

Also, not a fan of "newUrl" as the parameter in public API.